// validator/es/lib/alpha.js
export var alpha = {
  'en-US': /^[A-Z]+$/i,
  'az-AZ': /^[A-VXYZ]+$/i,
  'bg-BG': /^[-]+$/i,
  'cs-CZ': /^[A-Z]+$/i,
  'da-DK': /^[A-Z]+$/i,
  'de-DE': /^[A-Z]+$/i,
  'el-GR': /^[-]+$/i,
  'es-ES': /^[A-Z]+$/i,
  'fa-IR': /^[]+$/i,
  'fi-FI': /^[A-Z]+$/i,
  'fr-FR': /^[A-Z]+$/i,
  'it-IT': /^[A-Z]+$/i,
  'ja-JP': /^[----]+$/i,
  'nb-NO': /^[A-Z]+$/i,
  'nl-NL': /^[A-Z]+$/i,
  'nn-NO': /^[A-Z]+$/i,
  'hu-HU': /^[A-Z]+$/i,
  'pl-PL': /^[A-Z]+$/i,
  'pt-PT': /^[A-Z]+$/i,
  'ru-RU': /^[-]+$/i,
  'kk-KZ': /^[-\u04D8\u04B0\u0406\u04A2\u0492\u04AE\u049A\u04E8\u04BA]+$/i,
  'sl-SI': /^[A-Z]+$/i,
  'sk-SK': /^[A-Z]+$/i,
  'sr-RS@latin': /^[A-Z]+$/i,
  'sr-RS': /^[-]+$/i,
  'sv-SE': /^[A-Z]+$/i,
  'th-TH': /^[-\s]+$/i,
  'tr-TR': /^[A-Z]+$/i,
  'uk-UA': /^[-I]+$/i,
  'vi-VN': /^[A-Z]+$/i,
  'ko-KR': /^[---]*$/,
  'ku-IQ': /^[]+$/i,
  ar: /^[]+$/,
  he: /^[-]+$/,
  fa: /^['']+$/i,
  bn: /^['']+$/,
  eo: /^[ABCD-GHIJK-PRSTUVZ]+$/i,
  'hi-IN': /^[\u0900-\u0961]+[\u0972-\u097F]*$/i,
  'si-LK': /^[\u0D80-\u0DFF]+$/,
  'ta-IN': /^[\u0B80-\u0BFF]+$/i,
  'te-IN': /^[\u0C00-\u0C7F]+$/i,
  'kn-IN': /^[\u0C80-\u0CFF]+$/i,
  'ml-IN': /^[\u0D00-\u0D7F]+$/i,
  'gu-IN': /^[\u0A80-\u0AFF]+$/i,
  'pa-IN': /^[\u0A00-\u0A7F]+$/i,
  'or-IN': /^[\u0B00-\u0B7F]+$/i
};
export var alphanumeric = {
  'en-US': /^[0-9A-Z]+$/i,
  'az-AZ': /^[0-9A-VXYZ]+$/i,
  'bg-BG': /^[0-9-]+$/i,
  'cs-CZ': /^[0-9A-Z]+$/i,
  'da-DK': /^[0-9A-Z]+$/i,
  'de-DE': /^[0-9A-Z]+$/i,
  'el-GR': /^[0-9-]+$/i,
  'es-ES': /^[0-9A-Z]+$/i,
  'fi-FI': /^[0-9A-Z]+$/i,
  'fr-FR': /^[0-9A-Z]+$/i,
  'it-IT': /^[0-9A-Z]+$/i,
  'ja-JP': /^[0-9-----]+$/i,
  'hu-HU': /^[0-9A-Z]+$/i,
  'nb-NO': /^[0-9A-Z]+$/i,
  'nl-NL': /^[0-9A-Z]+$/i,
  'nn-NO': /^[0-9A-Z]+$/i,
  'pl-PL': /^[0-9A-Z]+$/i,
  'pt-PT': /^[0-9A-Z]+$/i,
  'ru-RU': /^[0-9-]+$/i,
  'kk-KZ': /^[0-9-\u04D8\u04B0\u0406\u04A2\u0492\u04AE\u049A\u04E8\u04BA]+$/i,
  'sl-SI': /^[0-9A-Z]+$/i,
  'sk-SK': /^[0-9A-Z]+$/i,
  'sr-RS@latin': /^[0-9A-Z]+$/i,
  'sr-RS': /^[0-9-]+$/i,
  'sv-SE': /^[0-9A-Z]+$/i,
  'th-TH': /^[-\s]+$/i,
  'tr-TR': /^[0-9A-Z]+$/i,
  'uk-UA': /^[0-9-I]+$/i,
  'ko-KR': /^[0-9---]*$/,
  'ku-IQ': /^[0-9]+$/i,
  'vi-VN': /^[0-9A-Z]+$/i,
  ar: /^[0-9]+$/,
  he: /^[0-9-]+$/,
  fa: /^['0-9']+$/i,
  bn: /^['']+$/,
  eo: /^[0-9ABCD-GHIJK-PRSTUVZ]+$/i,
  'hi-IN': /^[\u0900-\u0963]+[\u0966-\u097F]*$/i,
  'si-LK': /^[0-9\u0D80-\u0DFF]+$/,
  'ta-IN': /^[0-9\u0B80-\u0BFF.]+$/i,
  'te-IN': /^[0-9\u0C00-\u0C7F.]+$/i,
  'kn-IN': /^[0-9\u0C80-\u0CFF.]+$/i,
  'ml-IN': /^[0-9\u0D00-\u0D7F.]+$/i,
  'gu-IN': /^[0-9\u0A80-\u0AFF.]+$/i,
  'pa-IN': /^[0-9\u0A00-\u0A7F.]+$/i,
  'or-IN': /^[0-9\u0B00-\u0B7F.]+$/i
};
export var decimal = {
  'en-US': '.',
  ar: ''
};
export var englishLocales = ['AU', 'GB', 'HK', 'IN', 'NZ', 'ZA', 'ZM'];
for (var locale, i = 0; i < englishLocales.length; i++) {
  locale = "en-".concat(englishLocales[i]);
  alpha[locale] = alpha['en-US'];
  alphanumeric[locale] = alphanumeric['en-US'];
  decimal[locale] = decimal['en-US'];
}

// Source: http://www.localeplanet.com/java/
export var arabicLocales = ['AE', 'BH', 'DZ', 'EG', 'IQ', 'JO', 'KW', 'LB', 'LY', 'MA', 'QM', 'QA', 'SA', 'SD', 'SY', 'TN', 'YE'];
for (var _locale, _i = 0; _i < arabicLocales.length; _i++) {
  _locale = "ar-".concat(arabicLocales[_i]);
  alpha[_locale] = alpha.ar;
  alphanumeric[_locale] = alphanumeric.ar;
  decimal[_locale] = decimal.ar;
}
export var farsiLocales = ['IR', 'AF'];
for (var _locale2, _i2 = 0; _i2 < farsiLocales.length; _i2++) {
  _locale2 = "fa-".concat(farsiLocales[_i2]);
  alphanumeric[_locale2] = alphanumeric.fa;
  decimal[_locale2] = decimal.ar;
}
export var bengaliLocales = ['BD', 'IN'];
for (var _locale3, _i3 = 0; _i3 < bengaliLocales.length; _i3++) {
  _locale3 = "bn-".concat(bengaliLocales[_i3]);
  alpha[_locale3] = alpha.bn;
  alphanumeric[_locale3] = alphanumeric.bn;
  decimal[_locale3] = decimal['en-US'];
}

// Source: https://en.wikipedia.org/wiki/Decimal_mark
export var dotDecimal = ['ar-EG', 'ar-LB', 'ar-LY'];
export var commaDecimal = ['bg-BG', 'cs-CZ', 'da-DK', 'de-DE', 'el-GR', 'en-ZM', 'eo', 'es-ES', 'fr-CA', 'fr-FR', 'gu-IN', 'hi-IN', 'hu-HU', 'id-ID', 'it-IT', 'kk-KZ', 'kn-IN', 'ku-IQ', 'ml-IN', 'nb-NO', 'nl-NL', 'nn-NO', 'or-IN', 'pa-IN', 'pl-PL', 'pt-PT', 'ru-RU', 'si-LK', 'sl-SI', 'sr-RS', 'sr-RS@latin', 'sv-SE', 'ta-IN', 'te-IN', 'tr-TR', 'uk-UA', 'vi-VN'];
for (var _i4 = 0; _i4 < dotDecimal.length; _i4++) {
  decimal[dotDecimal[_i4]] = decimal['en-US'];
}
for (var _i5 = 0; _i5 < commaDecimal.length; _i5++) {
  decimal[commaDecimal[_i5]] = ',';
}
alpha['fr-CA'] = alpha['fr-FR'];
alphanumeric['fr-CA'] = alphanumeric['fr-FR'];
alpha['pt-BR'] = alpha['pt-PT'];
alphanumeric['pt-BR'] = alphanumeric['pt-PT'];
decimal['pt-BR'] = decimal['pt-PT'];

// see #862
alpha['pl-Pl'] = alpha['pl-PL'];
alphanumeric['pl-Pl'] = alphanumeric['pl-PL'];
decimal['pl-Pl'] = decimal['pl-PL'];

// see #1455
alpha['fa-AF'] = alpha.fa;
